internal/linker/link_node_modules.ts

```diff
@@ -276,36 +297,72 @@ export async function main(args: string[], runfiles: Runfiles) {
   // Now add symlinks to each of our first-party packages so they appear under the node_modules tree
   const links = [];
 
-  const linkModule =
-      async (name: string, root: LinkerRoot, modulePath: string) => {
-    let target: string = '<package linking failed>';
-    switch (root) {
-      case 'bin':
-        // FIXME(#1196)
-        target = path.join(workspaceAbs, bin, toWorkspaceDir(modulePath));
-        break;
-      case 'src':
-        target = path.join(workspaceAbs, toWorkspaceDir(modulePath));
-        break;
-      case 'runfiles':
-        target = runfiles.resolve(modulePath) || '<runfiles resolution failed>';
-        break;
-    }
+  const linkModules =
+      async (modules: {name: string, root: LinkerRoot, modulePath: string}[]) => {
+    for (const m of modules) {
+      let target: string = '<package linking failed>';
+      switch (m.root) {
+        case 'bin':
+          // FIXME(#1196)
+          target = path.join(workspaceAbs, bin, toWorkspaceDir(m.modulePath));
+          break;
+        case 'src':
+          target = path.join(workspaceAbs, toWorkspaceDir(m.modulePath));
+          break;
+        case 'runfiles':
+          target = runfiles.resolve(m.modulePath) || '<runfiles resolution failed>';
+          break;
+      }
 
-    await symlink(target, name);
+      // ensure the subdirectories exist
+      mkdirp(path.dirname(m.name));
```

Comment:
extra FS calls inside mkdirp right? how did we deal with scoped packages before (there must be a mkdir somewhere else already?)

Reply:
yes, there was a mkdirp already for the 2 segment case. for one segment module mappings path.dirname() will return `` and mkdirp is a noop.